4.5:raw-imu-logging opts and filter setup rewrite (#5709)
ArduPilot/ardupilot_wiki: common/source/docs/common-raw-imu-logging.rst

.. _common-raw-imu-logging:

======================================
Using Raw IMU Logging for FFT Analysis
======================================

Using the :ref:`IMU Batch Sampler<common-imu-batchsampling>` to gather data for spectral analysis of noise to setup notch filtering is the historical method and still valid. But with firmware 4.5 and later, and if using an H7 -based autopilot, a better method is available and can be combined with a new analysis and filter setup tool to simply the process by logging raw IMU data instead.

> The downside to this method is that it produces very large logs and unless the data gathering is restricted to a single gyro, some slower autopilots (F4-based for example) may not be able to keep up with resulting logging drop-outs). Also the data produced can only be analyzed by the `Notch Filter Review tool <https://firmware.ardupilot.org/Tools/WebTools/FilterReview/>`__ on the ArduPilot Firmware "Web Tools" page.

Reviewer: ```suggestion
The downside to this method is that it produces very large logs and unless the data gathering is restricted to a single gyro, some slower autopilots (F4-based for example may not be able to keep up with resulting logging drop-outs). Also the data produced can only be analyzed by the `Notch Filter Review tool <https://firmware.ardupilot.org/Tools/WebTools/FilterReview/>`__ on the ArduPilot Firmware "Web Tools" page.
`

However, using this greatly simplifies the notch filtering setup and a single test flight can usually be used to gather the data and the filters then be setup interactively with the Filter Review tool.

Filter Setup Process
====================

> #. Select the method that will be used for dynamically adjusting the notch(s) center frequency and do any required setup, ie if :ref:`ESC telemtry<common-esc-telemetry>` will be used make sure its setup, if Throttle based make sure your battery current sensor is calibrated and setup properly, etc.

Reviewer: ```suggestion
#. Select the method that will be used for dynamically adjusting the notch(s) center frequency and do any required setup, ie if :ref:`ESC telemtry<common-esc-telemetry>` will be used make sure its setup, if Throttle based battery voltage compensation should be setup.
```

Author: okay thinking of compass mot for some reason...this is throttle position based....changing..thanks

> #. :ref:`LOG_BITMASK <LOG_BITMASK>`:  Set the IMU_RAW bit (19). Since we will not be setting up :ref:`IMU batch sampling<common-imu-batchsampling>` (where short bursts of raw IMU data are gathered). But rather log the raw IMU data continuously.

Reviewer: You don't need to change `LOG_BITMASK` if your using `INS_RAW_LOG_OPT`.

Author: okay..must have missread your notes

Author: fixed

However, using this greatly simplifies the notch filtering setup and a single test flight can usually be used to gather the data and the filters then be setup interactively with the Filter Review tool.

Filter Setup Process
====================

#. Select the method that will be used for dynamically adjusting the notch(s) center frequency and do any required setup, ie if :ref:`ESC telemtry<common-esc-telemetry>` will be used make sure its setup, if Throttle based make sure your battery current sensor is calibrated and setup properly, etc.
#. :ref:`LOG_BITMASK <LOG_BITMASK>`:  Set the IMU_RAW bit (19). Since we will not be setting up :ref:`IMU batch sampling<common-imu-batchsampling>` (where short bursts of raw IMU data are gathered). But rather log the raw IMU data continuously.
#. To prevent overloading the logging, especially on slower autopilot CPUs, set :ref:`INS_RAW_LOG_OPT<INS_RAW_LOG_OPT>` bit 0 and bit 3 (value = "9"). This will restrict logging to the primary gyro and log both pre and post filtering.

> #. Perform a regular flight (not just a gentle hover) of at least 1 minute and :ref:`download the dataflash logs <common-downloading-and-analyzing-data-logs-in-mission-planner>`.

Reviewer: A gentle hover is still a good idea for a first pass, then a more realistic flight to double check. Or since you can split the log you can do a hover to start and then fly about more aggressively after. Because of the faster logging you also don't need a full min, 20 or 30 seconds should be plenty.

Author: done
